pennylane/ops/functions/dot.py

```diff
@@ -84,7 +99,28 @@ def dot(
         return ParametrizedHamiltonian(coeffs, ops)
 
     if pauli:
-        return _pauli_dot(coeffs, ops)
+        if all(isinstance(pauli, (PauliWord, PauliSentence)) for pauli in ops):
+            # Use pauli arithmetic when ops are just PauliWord and PauliSentence instances
+            return _dot_pure_paulis(coeffs, ops)
+
+        # Else, transform all ops to pauli sentences
+        return _dot_with_ops_and_paulis(coeffs, ops)
+
+    if any(isinstance(op, (PauliWord, PauliSentence)) for op in ops):
+        # promote any PauliWord/Sentence instances to operators
+        # Looks a bit complicated, mainly due to the fact of having so single out identities since they currently dont have an .operation()
+        # else we could just do ops = [op.operation() if isinstance(op, (PauliWord, PauliSentence)) else op for op in ops]
+        IdWord = PauliWord({})
+        new_ops = []
+        for op in ops:
+            if isinstance(op, (PauliWord, PauliSentence)):
+                if op == IdWord:
+                    new_ops.append(qml.Identity(0))
```

Comment:
I'm a bit hesitant about choosing the wire `0` for the identity.  We could potentially allow `Identity` to exist on "no wires" like with do with `GlobalPhase`. That might eliminate this problem of not knowing what wires to use here.

Reply:
That would be amazing!

Reply:
How to best proceed from here: Leave this PR open and blocked until that feature request is merged or continue with that as "fix-later"?

Reply:
I don't have the full context to know this, but do we want people to use `PauliWord({})`? If not, we don't really need to address this

Reply:
This is user facing. E.g. even when you initialize `PauliWord({0:"I"})` it will automatically strip all identity entries and you are left with `PauliWord({})` upon initialization. I think it makes sense that the empty PauliWord is 1 and the empty PauliSentence is 0.